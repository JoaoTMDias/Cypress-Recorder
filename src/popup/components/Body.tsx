import * as React from 'react';
import type { Block } from '../../types';
import { RecState } from '../../constants';
import CodeDisplay from './CodeDisplay';
import LandingBox from './LandingBox';
import Pulsar from './Pulsar';

export interface BodyProps {
  isValidTab: boolean,
  recStatus: RecState,
  codeBlocks: Block[],
  destroyBlock: (index: number) => void,
  moveBlock: (dragIdx: number, dropIdx: number) => void,
}

export default ({
  recStatus,
  codeBlocks,
  isValidTab,
  destroyBlock,
  moveBlock,
}: BodyProps) => {
  function renderContent() {
    switch (recStatus) {
      case RecState.OFF:
        return <LandingBox isValidTab={isValidTab} />;

      default:
        return codeBlocks && codeBlocks.length > 0 ? <CodeDisplay codeBlocks={codeBlocks} destroyBlock={destroyBlock} moveBlock={moveBlock} /> : <Pulsar />;
    }
  }

  return (
    <section className="body">
      {renderContent()}
    </section>
  );
};
