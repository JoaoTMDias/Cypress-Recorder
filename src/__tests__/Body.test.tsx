import * as React from 'react';
import { shallow } from 'enzyme';
import Body, { BodyProps } from '../popup/components/Body';
import LandingBox from '../popup/components/LandingBox';
import CodeDisplay from '../popup/components/CodeDisplay';
import { RecState } from '../constants';
import '../setupTests';

describe('Body', () => {
  let props: BodyProps;
  let wrapper;
  beforeAll(() => {
    props = {
      codeBlocks: [],
      isValidTab: true,
      destroyBlock: jest.fn(),
      moveBlock: jest.fn(),
      recStatus: RecState.OFF,
    };
  });
  it('Should render LandingBox when recStatus is off', () => {
    wrapper = shallow(<Body {...props} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.exists(LandingBox)).toBe(true);
    expect(wrapper.exists(CodeDisplay)).toBe(false);
  });
  it('Should render CodeDisplay when recStatus is paused and has codeBlocks', () => {
    props.recStatus = RecState.PAUSED;
    props.codeBlocks = [
      {
        id: "2XSrhOx7E",
        value: "cy.visit('https://review.feedzai.com/D27821#change-IxGsRMpnU7Uu');"
      }
    ];
    wrapper = shallow(<Body {...props} />);
    expect(wrapper.exists(CodeDisplay)).toBe(true);
    expect(wrapper.exists(LandingBox)).toBe(false);
  });
  it('Should render CodeDisplay when recStatus is on', () => {
    props.recStatus = RecState.ON;
    props.codeBlocks = [
      {
        id: "2XSrhOx7E",
        value: "cy.visit('https://review.feedzai.com/D27821#change-IxGsRMpnU7Uu');"
      }
    ];
    wrapper = shallow(<Body {...props} />);
    expect(wrapper.exists(CodeDisplay)).toBe(true);
    expect(wrapper.exists(LandingBox)).toBe(false);
  });
});
