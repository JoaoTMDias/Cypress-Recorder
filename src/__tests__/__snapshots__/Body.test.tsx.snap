// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body Should render LandingBox when recStatus is off 1`] = `
<section
  className="body"
>
  <Component
    isValidTab={true}
  />
</section>
`;
